pkg/cfn/builder/iam.go

```diff
@@ -525,6 +527,9 @@ func (rs *IAMRoleResourceSet) makeAssumeRolePolicyDocument() cft.MapOfInterfaces
 	}
 	if rs.serviceAccount != "" && rs.namespace != "" {
 		logger.Debug("service account location provided: %s/%s, adding sub condition", api.AWSNodeMeta.Namespace, api.AWSNodeMeta.Name)
+		if rs.subjectPattern != "" {
+			return rs.oidc.MakeAssumeRolePolicyDocumentWithServiceAccountConditionsAllowingWildcard(rs.namespace, rs.subjectPattern)
+		}
```

Comment:
Could we make ```MakeAssumeRolePolicyDocumentWithServiceAccountConditions``` take `rs.subjectPattern` as a parameter to avoid duplicating some of the logic